experimental/frame_sorter.go

```diff
@@ -0,0 +1,46 @@
+package experimental
+
+import "github.com/grafana/grafana-plugin-sdk-go/data"
+
+// NewFrameSorter - returns a new frameSorter
+func NewFrameSorter(frame *data.Frame, sortField *data.Field) FrameSorter {
+	return FrameSorter{frame, sortField}
+}
+
+// FrameSorter - sort a DataFrame by field
+type FrameSorter struct {
+	frame     *data.Frame
+	sortField *data.Field
+}
+
+func (fs FrameSorter) Len() int { return fs.frame.Rows() }
+func (fs FrameSorter) Swap(i, j int) {
+	for _, field := range fs.frame.Fields {
+		valA := field.At(i)
+		valB := field.At(j)
+		field.Set(j, valA)
+		field.Set(i, valB)
+	}
+}
+func (fs FrameSorter) Less(i, j int) bool {
+	switch kind := fs.sortField.Type(); kind {
+	case data.FieldTypeString:
+		valA := fs.sortField.At(i).(string)
+		valB := fs.sortField.At(j).(string)
+		return valA < valB
+	case data.FieldTypeNullableString:
+		valA := fs.sortField.At(i).(*string)
+		valB := fs.sortField.At(j).(*string)
+		return *valA < *valB
```

Comment:
This will panic if either of these are nil I believe

Reply:
@kylebrandt How about I address this issue and merge?  It's in the experimental package.  Addressing the other stuff (later) seems we could move it out of experimental.

Reply:
Check if nil and return true / false accordingly before the comparison, once both have been checked for in, then `*valA < *valB`  (dereferencing them) will be safe, because neither can be `nil` (as the code would have returned). Similar to how labels is treated it here https://github.com/grafana/grafana-plugin-sdk-go/blob/6807a8d3ceae94b2811a7fae3996f9c729568b5e/data/time_series.go#L556

Reply:
Agree with @kylebrandt.